### PR TITLE
update readme to include python benefits

### PR DESCRIPTION
**Pull request by dshcherb**
Some useful traits that I found about using python for charms as opposed
to using compiled languages.

Might be worthwhile to have that directly on the first page.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,9 +3,16 @@
 ## Overview
 
 CharmHelpers provides an opinionated set of tools for building Juju
-charms that work together. In addition to basic tasks like interact-
-ing with the charm environment and the machine it runs on, it also
-helps keep you build hooks and establish relations effortlessly.
+charms that work together. In addition to basic tasks like interacting
+with the charm environment and the machine it runs on, it also helps
+keep you build hooks and establish relations effortlessly.
```
Comment (ajkavanagh) [inline README.md]: This line doesn't read very well?  I'm not sure what it is saying.  Is it "... also helps you to build hooks and establish relations effortlessly"?

File: README.md [hunk 2]
```diff
@@ -3,9 +3,16 @@
...
+
+## Why Python?
+
+* Low entry barrier, fairly common in automation tools;
+* An interpreted language helps with charm portability across different CPU architectures;
+* Doesn't require debugging symbols (just use pdb in-place);
+* An author or a user is able to make debugging changes without recompiling a charm.
```
Comment (ajkavanagh) [inline README.md]: I would drop the first bullet point, and probably replace it with:

* Python is an extremely popular, easy to learn, and powerful language which is also common in automation tools.

File: README.md (hunk 1 above)
Comment (tvansteenburgh) [inline README.md]: s/keep//

Comment (dshcherb) [reply README.md]: Well, that was the original one - I just fixed formatting. I'll fix the comment itself as well.

**Comment by dshcherb** [comment]
Pull-request updated, HEAD is now 67687eeeb5a5ac14643a96cc82580e5372983170

**Pull request merged**